### ISSUE TITLE
Use mapnik-reference to validate datasourcer related parameters in MML

### ISSUE THREAD

**Issue by nebulon42** [mapbox/carto#480]
mapnik-reference defines which parameters can be used for which type of datasource. This information should be used to validate the parameters given for different layers in the MML file. The information could also be used for documentation of Mapnik datasources and their parameters.

**Comment by StyXman** [comment]
Related to this, today I pulled the latest master and I started getting this error:

```
./bin/carto ~/src/projects/osm/osm-carto/project.mml
Error: Invalid API version. A valid version is e.g. 3.0.0 or 3.0.10
```

I tried with some values:
* Some versions listed here: https://raw.githubusercontent.com/mapbox/carto/master/docs/mapnik_api.rst
* All versions from here (cut for brevity):
```
$ node
> mr = require('mapnik-reference');
{ version: 
   { '2.0.0': 
     '2.0.1': 
     '2.0.2':
     '2.1.0':
     '2.1.1':
     latest:
```

With all attemps I get:

```
Error: Version x.y.z is not supported
```

The default was removed in b0a03fa. I don't know what else should I do.

**Comment by nebulon42** [comment]
I cannot reproduce that. Are you sure that this is a vanilla master? You could try `rm -rf node_modules` and `npm install`. Maybe that helps.

**Comment by StyXman** [comment]
Yeah, that was it. So it means I updated a dependency. I wish the language would warn about those (but to be fair, Python does not do it either).
